Wrappers/Python/test/test_out_in_place.py

```diff
@@ -0,0 +1,309 @@
+            (KullbackLeibler(b=b, backend='numpy'), ag),
+            (L1Norm(), ag),
+            (L1Norm(), ig),
+            (L1Norm(b=b), ag),
+            (L1Norm(b=b, weight=b), ag),
+            (TranslateFunction(L1Norm(), b), ag),
+            (TranslateFunction(L2NormSquared(), b), ag),
+            (L2NormSquared(), ag),
+            (scalar * L2NormSquared(), ag),
+            (SumFunction(L2NormSquared(), scalar * L2NormSquared()), ag),
+            (SumScalarFunction(L2NormSquared(), 3), ag),
+            (ConstantFunction(3), ag),
+            (ZeroFunction(), ag),
+            (L2NormSquared(b=b), ag),
+            (L2NormSquared(), ag),
+            (LeastSquares(A, b_ig, c, weight_ls), ig),
+            (LeastSquares(A, b_ig, c), ig),
+            (WeightedL2NormSquared(weight=b_ig), ig),
+            (TotalVariation(backend='c', warm_start=False, max_iteration=100), ig),
+            (TotalVariation(backend='numpy', warm_start=False, max_iteration=100), ig),
+            (OperatorCompositionFunction(L2NormSquared(), A), ig),
+            (MixedL21Norm(), bg),
+            (SmoothMixedL21Norm(epsilon=0.3), bg),
+            (MixedL11Norm(), bg)
+
+        ]
+
+        np.random.seed(5)
+        self.data_arrays=[np.random.normal(0,1, (10,10)).astype(np.float32),  np.array(range(0,65500, 655), dtype=np.uint16).reshape((10,10)), np.random.uniform(-0.1,1,(10,10)).astype(np.float32)]
+
+    def get_result(self, function, method, x, *args):
+        try:
+            input=x.copy() #To check that it isn't changed after function calls 
+            if method == 'proximal':
+                out= function.proximal(x, *args)
+            elif method == 'proximal_conjugate':
+                out= function.proximal_conjugate(x, *args)
+            elif method == 'gradient':
+                out= function.gradient(x, *args)
+            self.assertDataArraysInContainerAllClose(input, x, rtol=1e-5, msg= "In case func."+method+'(data, *args) where func is  ' + function.__class__.__name__+ 'the input data has been incorrectly affected by the calculation. ')
+            return out
+        except NotImplementedError:
+            return None
+        
+    def in_place_test(self,desired_result, function, method,   x, *args, ): 
+            out3 = x.copy()
+            try:
+                if method == 'proximal':
+                    function.proximal(out3, *args, out=out3)
+                elif method == 'proximal_conjugate':
+                    function.proximal_conjugate(out3, *args, out=out3)
+                elif method == 'gradient':
+                    function.gradient(out3, *args, out=out3)
+                self.assertDataArraysInContainerAllClose(desired_result, out3, rtol=1e-5, msg= "In place calculation failed for func."+method+'(data, *args, out=data) where func is  ' + function.__class__.__name__+ '. ')
+      
+            except (InPlaceError, NotImplementedError):
+                pass
```

Comment:
Maybe a `logging.debug` of sort for `InPlaceError` would be useful.

I guess this is a thing we may want to have for most of the tests? @casperdcl ?

Reply:
At the moment, the tests are set so that they fail as soon as the test fails for one operator or function, it does not continue to run through the list. This made the code easier to read and more understandable